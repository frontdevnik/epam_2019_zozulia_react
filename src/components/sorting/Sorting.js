import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import SortByName from './SortByName';
import { sorByLikes, sortByRating, resetSorting } from './actions';
import style from './sorting.module.scss';
import withTranslation from '../../hocs/withTranslation';

function Sorting(props) {
  const { sorByLikes, sortByRating, resetSorting, sort_title, sort_by_likes, sort_by_rating, sort_reset, sort_search } = props;

  return (
    <div className={style.sorting}>
      <h3>{sort_title}</h3>
      <button type="button" onClick={sorByLikes}>{sort_by_likes}</button>
      <button type="button" onClick={sortByRating}>{sort_by_rating}</button>
      <button type="button" onClick={resetSorting}>{sort_reset}</button>
      <SortByName placeholder={sort_search} />
    </div>
  );
}

Sorting.propTypes = {
  sorByLikes: PropTypes.func.isRequired,
  sortByRating: PropTypes.func.isRequired,
  resetSorting: PropTypes.func.isRequired,
};

const mapDispatchToProps = ({
  sorByLikes,
  sortByRating,
  resetSorting,
});

const withConnect = connect(null, mapDispatchToProps);
const withTranslationWords = withTranslation(['sort_title', 'sort_by_likes', 'sort_by_rating', 'sort_reset', 'sort_search']);

export default withTranslationWords(withConnect(Sorting));
