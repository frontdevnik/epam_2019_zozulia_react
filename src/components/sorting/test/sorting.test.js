import React from 'react';
import {Sorting} from '../Sorting';
import {sortingProps} from './stubProps';
import createMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = createMockStore();

describe('<Sorting />', () => {
    let component;

    beforeEach(() => {
        component = mount(<Provider store={mockStore()}><Sorting {...sortingProps} /></Provider>);
    });

    it('should render correctly', () => {
        expect(component.html()).toMatchSnapshot();
    });

    it('should render title', () => {
        expect(component.find('h3').text()).toBe(sortingProps.sort_title);
    });

    it('should render sort by likes button', () => {
        expect(component.find('button').at(0).text()).toBe(sortingProps.sort_by_likes);
    });

    it('should call sortByLikes on click sort by likes button', () => {
        const button = component.find('button').at(0);
        button.simulate('click');
        expect(sortingProps.sorByLikes).toHaveBeenCalled();
    });

    it('should render sort by rating button', () => {
        expect(component.find('button').at(1).text()).toBe(sortingProps.sort_by_rating);
    });

    it('should call sortByRating on click sort by rating button', () => {
        const button = component.find('button').at(1);
        button.simulate('click');
        expect(sortingProps.sortByRating).toHaveBeenCalled();
    });

    it('should render reset sorting button', () => {
        expect(component.find('button').at(2).text()).toBe(sortingProps.sort_reset);
    });

    it('should call resetSorting on click reset sorting button', () => {
        const button = component.find('button').at(2);
        button.simulate('click');
        expect(sortingProps.resetSorting).toHaveBeenCalled();
    });
})
