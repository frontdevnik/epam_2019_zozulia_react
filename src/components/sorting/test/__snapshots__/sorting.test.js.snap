// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sorting /> should render correctly 1`] = `"<div class=\\"sorting\\"><h3>Sort movies: </h3><button type=\\"button\\">Sort by likes</button><button type=\\"button\\">Sort by rating</button><button type=\\"button\\">Reset filters</button><form class=\\"search\\"><button type=\\"submit\\" class=\\"search\\"></button><input name=\\"search\\" type=\\"search\\" placeholder=\\"Search\\" value=\\"\\"></form></div>"`;
